content/200-orm/800-more/300-upgrade-guides/200-upgrading-versions/500-upgrading-to-prisma-6.mdx

```diff
@@ -0,0 +1,312 @@
+---
+title: 'Upgrade to Prisma ORM 6'
+metaTitle: 'Upgrade to Prisma ORM 6'
+metaDescription: 'Guides on how to upgrade to Prisma ORM 6'
+tocDepth: 3
+toc: true
+---
+
+Prisma ORM v6 introduces a number of **breaking changes** when you upgrade from an earlier Prisma ORM version. This guide explains how this upgrade might affect your application and gives instructions on how to handle any changes.
+
+## Upgrade the `prisma` and `@prisma/client` packages to v6
+
+To upgrade to Prisma ORM v6 from an earlier version, you need to update both the `prisma` and `@prisma/client` packages:
+
+<TabbedContent code>
+
+<TabItem value="npm">
+
+```terminal
+npm install @prisma/client@6
+npm install -D prisma@6
+```
+
+</TabItem>
+
+<TabItem value="yarn">
+
+```terminal
+yarn up prisma@6 @prisma/client@6
+```
+
+</TabItem>
+
+<TabItem value="pnpm">
+
+```terminal
+pnpm upgrade prisma@6 @prisma/client@6
+```
+
+</TabItem>
+
+<TabItem value="bun">
+
+```terminal
+bun add @prisma/client@6
+bun add prisma@6 --dev
+```
+
+</TabItem>
+
+</TabbedContent>
+
+:::danger
+
+Before you upgrade, check each breaking change below to see how the upgrade might affect your application.
+
+:::
+
+
+## Breaking changes
+
+This section gives an overview of breaking changes in Prisma ORM v6.
+
+### Minimum supported Node.js versions
+
+The new minimum supported Node.js version sfor Prisma ORM v6 are:
+
+- for Node.js 18 the minimum supported version is **18.18.0**
+- for Node.js 20 the minimum supported version is **20.9.0**
+- for Node.js 22 the minimum supported version is **22.11.0**
+
+There is _no_ official support for Node.js 16, 17, 19 and 21.
+
+### Minimum supported TypeScript version
+
+The new minimum supported TypeScript version for Prisma ORM v6 is: **5.1.0**.
+
+### Schema change for implicit m-n relations on PostgreSQL
+
+If you're using PostgreSQL and are defining [implicit many-to-many relations](/orm/prisma-schema/data-model/relations/many-to-many-relations#implicit-many-to-many-relations) in your Prisma schema, Prisma ORM maintains the [relation table](/orm/prisma-schema/data-model/relations/many-to-many-relations#relation-tables) for you under the hood. This relation table has `A` and `B` columns to represent the tables of the models that are part of this relation.
+
+
+Previous versions of Prisma ORM used to create a _unique index_ on these two columns. In Prisma v6, this unique index is changing to a _primary key_ in order to [simplify for the default replica identity behaviour](https://github.com/prisma/prisma/issues/25196).
+
+<details>
+<summary>Expand for an example</summary>
+
+As an example, consider this relation table of an implicit m-n relation between `Post` and `Tag` models:
+
```

Comment:
Can we please have a `schema.prisma` that includes `Post` and `Tag` models, from which we derived this SQL migration? Without that, this section feels confusing, even with the link to https://www.prisma.io/docs/orm/prisma-schema/data-model/relations/many-to-many-relations#implicit-many-to-many-relations that is posted above.